src/uint256.h

```diff
@@ -106,6 +108,25 @@ std::optional<uintN_t> FromHex(std::string_view str)
     rv.SetHexDeprecated(str);
     return rv;
 }
+/**
+ * @brief Like FromHex(std::string_view str), but allows an "0x" prefix
+ *        and pads the input with leading zeroes if it is shorter than
+ *        the expected length of uintN_t::size()*2.
+ *
+ *        Designed to be used when dealing with user input.
+ */
+template <class uintN_t>
+std::optional<uintN_t> FromUserHex(std::string_view input)
```

Comment:
Since this is a "User" centric specialization of `FromHex` (and not a `Hex` specialization of `FromUser`), I'd keep the `FromHex` prefix, i.e.: `FromHexUser` or perhaps `FromHexLenient`, since the `User` part is just incidental

Reply:
I have nothing against any of the suggested names, but personally I still find `FromUserHex` to be the most intuitive (and it seems 2 at least 2 other reviewers like it as well), so I'm going to stick with that if that's okay?